core/src/main/java/nu/yona/server/subscriptions/service/UserService.java

```diff
@@ -462,17 +462,31 @@ public UserDto updateUser(UUID id, UserDto user)
 		Optional<ConfirmationCode> confirmationCode = createConfirmationCodeIfNumberUpdated(user, originalUser,
 				updatedUserEntity);
 		User savedUserEntity = userRepository.save(updatedUserEntity);
-		UserDto userDto = createUserDtoWithPrivateData(savedUserEntity);
 		if (confirmationCode.isPresent())
 		{
 			sendConfirmationCodeTextMessage(updatedUserEntity.getMobileNumber(), confirmationCode.get(),
 					SmsTemplate.CHANGED_USER_NUMBER_CONFIRMATION);
 		}
+		UserDto userDto = createUserDtoWithPrivateData(savedUserEntity);
 		logger.info("Updated user with mobile number '{}' and ID '{}'", userDto.getMobileNumber(), userDto.getId());
 		buddyService.broadcastUserInfoChangeToBuddies(savedUserEntity, originalUser);
 		return userDto;
 	}
 
+	@Transactional
+	public UserDto updateUserPhoto(UUID id, Optional<UUID> userPhotoId)
+	{
+		User userEntity = getUserEntityById(id);
+		UserDto originalUser = createUserDtoWithPrivateData(userEntity);
+		userEntity.setUserPhotoId(userPhotoId);
+		User savedUserEntity = userRepository.save(userEntity);
+		UserDto userDto = createUserDtoWithPrivateData(savedUserEntity);
+		logger.info("Updated user photo for user with mobile number '{}' and ID '{}'", userDto.getMobileNumber(),
```

Comment:
Couldn't it be helpful to know whether the user updated their photo or removed it?

Reply:
Will change logging